### PR TITLE
Metadata schema

### PR DESCRIPTION
**Pull request by bergie**
Add definitions for commonly used metadata keys in graph JSON schema

### CODE REVIEW

File: schema/graph.json [hunk 1]
```diff
@@ -13,7 +13,35 @@
...
-          "type": "string"
+          "type": "string",
+          "description": "Name of the graph"
+        },
+        "environment": {
+          "type": "object",
+          "description": "Information about the execution environment for the graph",
+          "additionalProperties": true,
+          "required": [
+            "type"
```
Comment (jonnor) [inline schema/graph.json]: `type` cannot be mandatory? Many existing graphs don't have it, and for instance `fbp` does not provide it either unless it is in the .fbp input file

File: schema/graph.json [hunk 2]
```diff
@@ -13,7 +13,35 @@
...
+            }
+          }
+        },
+        "description": {
+          "type": "string",
+          "description": "Graph description"
+        },
+        "icon": {
+          "type": "string",
+          "description": "Icon that can be used for depicting the graph"
```
Comment (jonnor) [inline schema/graph.json]: Should be an icon *name* right? Preferably/normally one of the Font Awesome names

File: schema/graph.json [hunk 3]
```diff
@@ -155,7 +219,22 @@
           "data": {},
           "metadata": {
             "type": "object",
-            "additionalProperties": true
+            "additionalProperties": true,
+            "properties": {
+              "route": {
+                "type": "integer",
+                "description": "Route identifier of a graph edge"
```
Comment (jonnor) [inline schema/graph.json]: What is this identifier used for? Why is it a number? What does the number refer?

File: schema/graph.json (hunk 1 above)
Comment (bergie) [reply schema/graph.json]: @jonnor I set it as required _if_ you have `environment` object (which itself is not required)

File: schema/graph.json (hunk 3 above)
Comment (bergie) [reply schema/graph.json]: The route number is used for edge color

File: schema/graph.json (hunk 2 above)
Comment (bergie) [reply schema/graph.json]: Fixed

File: schema/graph.json (hunk 1 above)
Comment (jonnor) [reply schema/graph.json]: OK

**Pull request merged**